src/dashboard/apps/consent/helpers.py

```diff
@@ -76,3 +83,65 @@ def send_mail(recipients: list, template_id: int, email_data: dict) -> int | Non
         # fail silently and send a sentry log
         sentry_sdk.capture_exception(e)
         return 0
+
+
+def duplicate_expiring_consents():
+    """Duplicate expiring consents to ensure continuity.
+
+    Perform the duplication of expiring consents to ensure continuity for consents that
+    are nearing their expiration date and have not been duplicated yet.
+    The function identifies consents that are set to expire shortly and generates
+    new consents starting immediately after the expiry date of the current consent,
+    with an extended validity period.
+
+    - Consents should be duplicated if their end date is earlier than or equal to
+    `now + CONSENT_NUMBER_DAYS_END_DATE`.
+    - duplicated consent date:
+        - start date of new consent is end date of current consent
+        - end date of new consent is calculate from CONSENT_NUMBER_DAYS_END_DATE
+
+    Returns:
+        List[Consent]: A list of newly created Consent objects that have been duplicated
+         from the expiring consents.
+    """
+    # calculate the date from which to check
+    checking_date = consent_checking_date()
+
+    # calculate the end date of new duplicate consents
+    duplicate_end_date = duplicate_consent_end_date()
+
+    # Subquery to check if a duplicate consent exists for the active consent.
+    # Only returns the id of the consents found for verification.
+    duplicate_consents = Consent.objects.filter(
+        start__gte=OuterRef("end"),
+        delivery_point=OuterRef("delivery_point"),
+        id_station_itinerance=OuterRef("id_station_itinerance"),
+        station_name=OuterRef("station_name"),
+        provider_assigned_id=OuterRef("provider_assigned_id"),
+    ).values("provider_assigned_id")
+
+    # Filter consents expiring before the checking date
+    expiring_consents = Consent.active_objects.exclude(
```

Comment:
I think we must restrict on validated consents instead:

```suggestion
    expiring_consents = Consent.validated_objects.exclude(
```